contracts/0.8.9/WithdrawalQueueBase.sol

```diff
@@ -0,0 +1,524 @@
+        uint256 amountOfStETH,
+        uint256 amountOfShares
+    );
+    event WithdrawalBatchFinalized(
+        uint256 indexed from, uint256 indexed to, uint256 amountOfETHLocked, uint256 sharesBurned, uint256 timestamp
+    );
+    event WithdrawalClaimed(
+        uint256 indexed requestId, address indexed owner, address indexed receiver, uint256 amountOfETH
+    );
+
+    error ZeroAmountOfETH();
+    error ZeroShareRate();
+    error ZeroTimestamp();
+    error ZeroRecipient();
+    error InvalidOwner(address _owner, address _sender);
+    error InvalidOwnerAddress(address _owner);
+    error InvalidRequestId(uint256 _requestId);
+    error InvalidRequestIdRange(uint256 startId, uint256 endId);
+    error NotEnoughEther();
+    error RequestNotFinalized(uint256 _requestId);
+    error RequestAlreadyClaimed(uint256 _requestId);
+    error InvalidHint(uint256 _hint);
+    error CantSendValueRecipientMayHaveReverted();
+
+    /// @notice id of the last request.
+    function getLastRequestId() public view returns (uint256) {
+        return LAST_REQUEST_ID_POSITION.getStorageUint256();
+    }
+
+    /// @notice id of the last finalized request
+    function getLastFinalizedRequestId() public view returns (uint256) {
+        return LAST_FINALIZED_REQUEST_ID_POSITION.getStorageUint256();
+    }
+
+    /// @notice amount of ETH on this contract balance that is locked for withdrawal and waiting for claim
+    function getLockedEtherAmount() public view returns (uint256) {
+        return LOCKED_ETHER_AMOUNT_POSITION.getStorageUint256();
+    }
+
+    /// @notice length of the checkpoints. Last possible value for the claim hint
+    function getLastCheckpointIndex() public view returns (uint256) {
+        return LAST_CHECKPOINT_INDEX_POSITION.getStorageUint256();
+    }
+
+    /// @notice return the number of unfinalized requests in the queue
+    function unfinalizedRequestNumber() external view returns (uint256) {
+        return getLastRequestId() - getLastFinalizedRequestId();
+    }
+
+    /// @notice Returns the amount of stETH inthe queue yet to be finalized
+    function unfinalizedStETH() external view returns (uint256) {
+        return
+            _getQueue()[getLastRequestId()].cumulativeStETH - _getQueue()[getLastFinalizedRequestId()].cumulativeStETH;
+    }
+
+    /// @notice Returns all withdrawal requests placed for the `_owner` address
+    ///
+    /// WARNING: This operation will copy the entire storage to memory, which can be quite expensive. This is designed
+    /// to mostly be used by view accessors that are queried without any gas fees. Developers should keep in mind that
+    /// this function has an unbounded cost, and using it as part of a state-changing function may render the function
+    /// uncallable if the set grows to a point where copying to memory consumes too much gas to fit in a block.
+    function getWithdrawalRequests(address _owner) external view returns (uint256[] memory requestsIds) {
```

Comment:
maybe + paginated version?

Reply:
It's relatively useless, because EnumeratedSet does not guarantee the ordering. So requests can shift between the calls and can be missed.